mochify/lib/poll-events.js

```diff
@@ -2,23 +2,32 @@
 
 exports.pollEvents = pollEvents;
 
-async function pollEvents(driver, emit) {
-  for (;;) {
-    const events = await driver.evaluateReturn('mocha.mochify_pollEvents()');
-    if (!events) {
-      continue;
-    }
-    for (const [event, data] of events) {
-      if (event === 'mochify.callback') {
-        return data.code || 0; // stop polling
+function pollEvents(driver, emit) {
+  return new Promise((resolve) => {
+    async function doPoll() {
+      const events = await driver.evaluateReturn('mocha.mochify_pollEvents()');
```

Comment:
One thing I do not understand in the current architecture yet: why does a driver need to expose a returning and a non-returning `evaluate`? Couldn't `evaluate` always return and then the caller can either discard or use the value?

Reply:
That is how I did it first, but I couldn't get it to work with driver-webdriver. It wraps the passed string in an anonymous function, and Puppeteer doesn't. If you find out how we can get away with a single function, I'm all for it.